monotime/nanotime_test.go

```diff
@@ -22,3 +23,18 @@ func TestNow(t *testing.T) {
 		}
 	}
 }
+
+func TestSince(t *testing.T) {
+	for i := 0; i < 100; i++ {
+		t1 := Now()
+		time.Sleep(1)
+		dur := Since(t1)
+		if dur <= 0 {
```

Comment:
This is flakey. I'm not exactly sure why, but at least on my Mac I have seen this fail when running this test 500,000 times:
```
$ go test -count 500000
--- FAIL: TestSince (0.00s)
	nanotime_test.go:33: dur=-56ns should have been greater than 0
--- FAIL: TestNow (0.00s)
	nanotime_test.go:22: t1=1484076828022408583 should have been less than or equal to t2=1484076828022408000
--- FAIL: TestSince (0.00s)
	nanotime_test.go:33: dur=-193ns should have been greater than 0
FAIL
exit status 1
FAIL	github.com/aristanetworks/goarista/monotime	31.487s
```

I have observed in the past that on Darwin systems, the monotime code will give you a value with microsecond precision instead of nanosecond.